src/TaylorBot.Net/Program.Commands.Discord/src/TaylorBot.Net.Commands.Discord.Program/Modules/DiscordInfo/Commands/OptionalAvatarTypeParser.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Text.Json;
+using System.Threading.Tasks;
+using OperationResult;
+using TaylorBot.Net.Commands.Parsers;
+using static OperationResult.Helpers;
+
+namespace TaylorBot.Net.Commands.Discord.Program.Modules.DiscordInfo.Commands
+{
+    public enum AvatarType
+    {
+        Guild,
+        Global,
+    }
+    
+    public class OptionalAvatarTypeParser : IOptionParser<AvatarType?>
+    {
+        public static Result<AvatarType, ParsingFailed> Parse(string input)
```

Comment:
The separate method is cleaner but I would put it `private`! It was public in the Last.fm parser because it needed to be used by the prefix command